### PR TITLE
fix(apigateway): move resource api_gateway_group into services and fix codecheck errors

### PR DESCRIPTION
**Pull request by wangyuancheng01**
**What this PR does / why we need it**:  move resource api_gateway_group into services and fix codecheck errors

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/apigateway" TESTARGS="-run TestAccApiGatewayGroup_basic"
...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apigateway -v -run TestAccApiGatewayGroup_basic -timeout 360m -parallel 4
=== RUN   TestAccApiGatewayGroup_basic
=== PAUSE TestAccApiGatewayGroup_basic
=== CONT  TestAccApiGatewayGroup_basic
--- PASS: TestAccApiGatewayGroup_basic (22.51s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apigateway        22.545s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/apigateway/resource_huaweicloud_api_gateway_group_test.go [hunk 1]
```diff
@@ -102,17 +102,17 @@ func testAccCheckApiGatewayGroupExists(n string) resource.TestCheckFunc {
 func testAccApigwGroup_basic(rName string) string {
 	return fmt.Sprintf(`
 resource "huaweicloud_api_gateway_group" "acc_apigw_group" {
-	name = "%s"
-	description = "created by acc test"
+  name = "%s"
+  description = "created by acc test"
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/apigateway/resource_huaweicloud_api_gateway_group_test.go]:
```suggestion
  name        = "%s"
  description = "created by acc test"
```

File: huaweicloud/services/acceptance/apigateway/resource_huaweicloud_api_gateway_group_test.go [hunk 2]
```diff
@@ -102,17 +102,17 @@ func testAccCheckApiGatewayGroupExists(n string) resource.TestCheckFunc {
...
 `, rName)
 }
 
 func testAccApigwGroup_update(rNameUpdate string) string {
 	return fmt.Sprintf(`
 resource "huaweicloud_api_gateway_group" "acc_apigw_group" {
-	name = "%s"
-	description = "updated by acc test"
+  name = "%s"
+  description = "updated by acc test"
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/apigateway/resource_huaweicloud_api_gateway_group_test.go]:
```suggestion
  name        = "%s"
  description = "updated by acc test"
```

File: huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_group.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	apigwClient, err := cfg.ApiGatewayV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating api gateway client: %s", err)
+	}
+
+	if err := groups.Delete(apigwClient, d.Id()).ExtractErr(); err != nil {
+		return common.CheckDeletedDiag(d, err, "api apis")
+	}
+
+	d.SetId("")
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_group.go]:
```suggestion
```

File: huaweicloud/services/acceptance/apigateway/resource_huaweicloud_api_gateway_group_test.go (hunk 1 above)
Comment (wangyuancheng01) [reply huaweicloud/services/acceptance/apigateway/resource_huaweicloud_api_gateway_group_test.go]: down

File: huaweicloud/services/acceptance/apigateway/resource_huaweicloud_api_gateway_group_test.go (hunk 2 above)
Comment (wangyuancheng01) [reply huaweicloud/services/acceptance/apigateway/resource_huaweicloud_api_gateway_group_test.go]: down

File: huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_group.go (hunk 3 above)
Comment (wangyuancheng01) [reply huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_group.go]: down

**Pull request merged**